src/core/net/dns_client.cpp

```diff
@@ -1556,16 +1557,41 @@ void Client::HandleTimer(void)
 
 #if OPENTHREAD_CONFIG_DNS_CLIENT_NAT64_ENABLE
 
-Error Client::ReplaceWithIp4Query(Query &aQuery)
+Error Client::ReplaceWithIp4Query(Query &aQuery, const Message &aResponseMessage)
 {
     Error     error = kErrorFailed;
     QueryInfo info;
+    Header    header;
 
     info.ReadFrom(aQuery);
 
     VerifyOrExit(info.mQueryType == kIp6AddressQuery);
     VerifyOrExit(info.mConfig.GetNat64Mode() == QueryConfig::kNat64Allow);
 
+    // Check the response to the IPv6 query from the server. If the
+    // response code is success but the answer section is empty
+    // (indicating the name exists but has no IPv6 address), or the
+    // response code indicates an error other than `NameError`, we
+    // replace the query with an IPv4 address resolution query for
+    // the same name. If the server responded with `NameError`
+    // (RCode=3), it indicates that the name doesn't exist, so there
+    // is no need to try an IPv4 query.
+
+    SuccessOrExit(aResponseMessage.Read(aResponseMessage.GetOffset(), header));
+
+    switch (header.GetResponseCode())
+    {
+    case Header::kResponseSuccess:
+        VerifyOrExit(header.GetAnswerCount() == 0);
+        OT_FALL_THROUGH;
+
+    default:
```

Comment:
So far I've always seen the 'default' case placed at the end! Is there any style guidance on this, or a particular reason to place it here?

Reply:
Generally, all `case` options, (including `default`) in a `switch()` statement can be in any order.

I generally follow the model where have the `case` where we `ExitNow()` be at the end. Some toolchains (code checking tools) require a `break` (or `OT_FALL_THROUGH`) when the `case` is not the last one. Since that `break` would follow an `ExitNow()` statement, it can be flaged it as "unreachable code" by some code analyzers. This means to address it all, we may need to do something like this:

```cpp
    case Header::kResponseNameError:
        ExitNow();
        OT_UNRECHABLE_CODE(break);

     case ...
```

which is somewhat ugly. Having the `case` that does `ExitNow()` at the end avoids this.